flowgate-api/src/main/java/com/vmware/flowgate/service/AssetService.java

```diff
@@ -210,6 +221,49 @@ public boolean isAssetNameValidate(String assetName) {
       return assetNames;
    }
 
+   public List<AssetIPMapping> batchCreateMappingFromFile(MultipartFile multipartFile)
+         throws IOException {
+      List<AssetIPMapping> failureMappings = new ArrayList<AssetIPMapping>();
+      InputStream inputStream = multipartFile.getInputStream();
+      try (InputStreamReader inputStreamReader = new InputStreamReader(inputStream, "UTF-8");
+            BufferedReader br = new BufferedReader(inputStreamReader);) {
+         String assetIPMappingString = null;
+         while ((assetIPMappingString = br.readLine()) != null) {
+            AssetIPMapping mapping = null;
+            if(assetIPMappingString.contains(TAB)) {
+               String contentsArray[] = assetIPMappingString.trim().split(TAB);
+               mapping = generateAssetIPMapping(contentsArray);
+            }else {
+               String contentsArray[] = assetIPMappingString.trim().split(SPACE);
+               mapping = generateAssetIPMapping(contentsArray);
+            }
+            if(isAssetNameValidate(mapping.getAssetname()) && IPAddressUtil.isValidIp(mapping.getIp())) {
+               BaseDocumentUtil.generateID(mapping);
+               assetIPMappingRepository.save(mapping);
+            }else {
+               failureMappings.add(mapping);
+            }
+         }
+      }
+      return failureMappings;
+   }
+
+   public static AssetIPMapping generateAssetIPMapping(String contentsArray[]) {
+      AssetIPMapping mapping = new AssetIPMapping();
+      for(String content : contentsArray) {
```

Comment:
for 

!StringUtils.isempty(x){
if(mapping.ip is empty
}

Reply:
Done